### ISSUE TITLE
Bambusa / Expired demos aren't deleted

### ISSUE THREAD

**Issue by chillu** [silverstripe/bambusa-installer#82]
Moved from https://github.com/silverstripeltd/silverstripe.com/issues/126 to simplify backlog planning

#### TL;DR; looks like we don't trigger platform API

Here's another demo that didn't get deleted:
 - serge-05
 - https://platform.silverstripe.com/naut/project/ssdemo/environment/prod/demo/224

## Description

I've received the email (as expected):

```
Your personal SilverStripe demo https://serge04-sscom.demo.silverstripe.cc
will expire after 13 September 2019. Simply login into your demo before
it expires to extend it by 7 days.

Once expired, you will no longer be able to access the demo.
Any content within the demo will be permanently removed.

The SilverStripe team
```

Hovewer, today (`2019-09-19`) I'm still seeing it existing in the cluster.

Here's what Platform API returned to me:
```json
http https://platform.silverstripe.com/naut/project/ssdemo/environment/prod/demo/223
HTTP/1.1 200 OK
Cache-Control: no-cache, no-store, must-revalidate
Connection: keep-alive
Content-Length: 1477
Content-Security-Policy: default-src 'none'; connect-src 'self' https://wchat.freshchat.com https://www.google-analytics.com; font-src 'self' https://use.typekit.net https://fonts.gstatic.com https://maxcdn.bootstrapcdn.com; frame-src 'self' https://silverstripe.webpush.freshchat.com https://wchat.freshchat.com; img-src 'self' data: https://p.typekit.net https://platform.slack-edge.com https://www.googletagmanager.com https://www.google-analytics.com; script-src 'self' 'unsafe-eval' 'unsafe-inline' https://use.typekit.net https://wchat.freshchat.com https://www.googletagmanager.com https://www.google-analytics.com https://ssl.google-analytics.com; style-src 'self' 'unsafe-inline' https://wchat.freshchat.com https://use.typekit.net https://maxcdn.bootstrapcdn.com https://fonts.googleapis.com; manifest-src 'self'; base-uri 'self'; form-action 'self' https://*.platform.silverstripe.com https://*.cwp.govt.nz https://silverstripe.freshdesk.com https://slack.com; frame-ancestors 'self'
Content-Type: application/vnd.api+json
Date: Wed, 18 Sep 2019 23:22:14 GMT
Last-Modified: Wed, 18 Sep 2019 23:22:09 GMT
Referrer-Policy: same-origin
Server: nginx
Strict-Transport-Security: max-age=31536000
Vary: X-Forwarded-Protocol
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-XSS-Protection: 1; mode=block

{
    "data": {
        "attributes": {
            "admin_password": "2ZnpcUyAQE0nxPDv",
            "admin_username": "admin",
            "date_expire_unix": 0,
            "endpoint": "https://serge04-sscom.demo.silverstripe.cc",
            "image": "silverstripe/bambusa-installer:latest",
            "name": "serge04-sscom",
            "status": "Finished"
        },
        "id": 223,
        "links": {
            "self": "https://platform.silverstripe.com/naut/project/ssdemo/environment/prod/demo/223"
        },
        "relationships": {
            "environment": {
                "data": {
                    "id": "prod",
                    "links": {
                        "self": "https://platform.silverstripe.com/naut/project/ssdemo/environment/prod"
                    },
                    "type": "environments"
                }
            },
            "snapshot": {
                "data": {
                    "id": 21718,
                    "links": {
                        "self": "https://platform.silverstripe.com/naut/project/ssdemo/snapshots/21718"
                    },
                    "meta": {
                        "url": "https://platform.silverstripe.com/naut/project/ssdemo/snapshots/21718"
                    },
                    "type": "snapshots"
                }
            }
        },
        "type": "demos"
    },
    "meta": {
        "server_time": "1568848934",
        "status_code": 200
    }
}
```

When I manually triggered the DELETE for the demo it worked just fine.
Now I can see the demo pod is actually removed from the cluster and the platform API returns 404 for it. Meaning - the problem is on the `ss.com` side.

I've had a look around Raygun logs - nothing reported in there. Looks like tho job didn't trigger the deletion.
